src/app/inventory/store/energy.ts

```diff
@@ -30,22 +29,18 @@ export function energyUpgrade(
   const tierSocket =
     item.sockets && getFirstSocketByCategoryHash(item.sockets, SocketCategoryHashes.ArmorTier);
 
-  if (!tierSocket?.socketDefinition.reusablePlugSetHash) {
+  if (!tierSocket?.plugSet) {
     return [];
   }
 
-  const plugSet = defs.PlugSet.get(tierSocket.socketDefinition.reusablePlugSetHash);
-
   const energyMods: DestinyInventoryItemDefinition[] = [];
-  for (const { plugItemHash } of plugSet.reusablePlugItems) {
-    const plugItem = defs.InventoryItem.get(plugItemHash);
-
-    const capacity = plugItem.plug?.energyCapacity;
-    if (!plugItem.plug || !capacity) {
+  for (const dimPlug of tierSocket.plugSet.plugs) {
+    const capacity = dimPlug.plugDef.plug?.energyCapacity;
```

Comment:
i think if you check types this optional chain is no longer necessary

Reply:
Done